### DISCUSSION TITLE
Doubts regarding Explicit Time Integrators.

### DISCUSSION THREAD

**Discussion by zuckarm** [idaholab/moose#29044] [Q&A Modules: Solid mechanics]
hi all, 
i had a question about explicit midpoint scheme, which is for first order odes. is there a specific way to specify this time integrator system, as i am not seeing the desired solution when i use this scheme. 

when i use central difference scheme, i am getting the desired solution.

**Comment by GiudGiud** [comment]
> which is for first order odes. is there a specific way to specify this time integrator system, as i am not seeing the desired solution when i use this scheme.

I dont think so. There are a few options you can select, notably for preconditioning
https://mooseframework.inl.gov/source/timeintegrators/ActuallyExplicitEuler.html

> when i use central difference scheme, i am getting the desired solution.

That one is second order. But you have to set `solve_type = LINEAR`

**Reply by zuckarm** [reply]
i tried using ActuallyExplicitEuler, but i am facing the following error:
```
*** ERROR ***
The following error occurred in the TimeStepper 'ConstantDT' of type ConstantDT.

Solve failed and timestep already at or below dtmin, cannot continue!
```
i have reduced dtmin to 1e-50, still the same error.

i have solved dynamic problem using both implicit and explicit methods. following are the plots of displacement vs time for different methods. i am not able to figure what parameters need to changed in the code for first-order methods.

this plot is using implicit methods:
[
![implicit_methods_disp_plot3](https://github.com/user-attachments/assets/1fb1d55d-723f-43f5-b768-a9d9366815cc)
](url)

this plot is using explicit methods:
![explicit_methods_disp_plot3](https://github.com/user-attachments/assets/0a532505-ef91-4187-bd57-5ba7ca8da072)

**Reply by GiudGiud** [reply]
> i tried using ActuallyExplicitEuler, but i am facing the following error:

This error means the solve did not converge. You can try these techniques to make it converge:
https://mooseframework.inl.gov//application_usage/failed_solves.html

> i have solved dynamic problem using both implicit and explicit methods. following are the plots of displacement vs time for different methods. i am not able to figure what parameters need to changed in the code for first-order methods.

You mostly need to reduce the time step for 1st order methods. 2nd order can use larger time steps to get the right solution.

Not every time integration method will work for every problem as I am sure you are aware. Central difference for example is unstable for advection problems with higher Peclet numbers
